### ISSUE TITLE
donkeyGallery installation problem

### ISSUE THREAD

**Issue by gsalihk** [Gix075/donkeyGallery#2]
on my localhost, I copied dnk-gallery folder and also demo page.when i tried to open the demo page, what I got is a page showing that the galleries are about to be downloaded but never happen. circles are forever circling.

what could be wrong?

![donkeygallery](https://cloud.githubusercontent.com/assets/10588263/5794264/0470483a-9f6f-11e4-8a60-1c999e1af18d.png)

**Comment by Gix075** [comment]
Hi gsalihk , for first i had to say that i never test donkey gallery on windows wamp (I'm a Mac/Linux user), so i don't know if it can works well on that environment. So it's difficult to say what is wrong without more info. It could be a system problem (directory write permission), or php or javascript problem. For first try to set all donkey gallery folder to write permission enabled. If it doesn't work try to read the console to understand if is a javascript error. Let me know ... I will try to help you!

**Comment by jamminjames** [comment]
It does the same thing when opening gallery-demo.html in my Firefox or Chrome browser, not using localhost. Same result when I upload the donkeyGallery-master folder and all contents onto a live server. What's wrong?

**Comment by Gix075** [comment]
I can't help you without more information about it. Are you sure that your folder has the right write permissions?

**Comment by jamminjames** [comment]
Yes, the folder permissions are ok. Considering it's not working in all these situations, I think maybe something's messed up with the code in the gallery-demo.

**Comment by Gix075** [comment]
For me this file works well in 7 different servers where i tested it. I can't help you without more informations, are you on Linux environment, there are some console error, can you say me which hosting provider you are using?

**Comment by jamminjames** [comment]
What works well on 7 servers? A full installation, or the gallery-demo, or both? My website is on a Linux environment. It's a dedicated server with a small company called Omsoft.

**Comment by Gix075** [comment]
For me work well both. I want help you, but it's no simple from here. Can you check if there are some console error during page loading?

**Comment by jamminjames** [comment]
When using Inspect Element in Firefox, console tab, it shows "Type error: data is null" for jquery.donkeyGallery.js, line 6, character 20. http://www.humortimes.com/wp-content/mu-plugins/donkeyGallery-master/gallery-demo.html

**Comment by Gix075** [comment]
Your installation is in a subdir so you need to set the **subdomain: ""** parameter (see the documentation to learn more about it) and try again with subdomain: "/wp-content/mu-plugins/donkeyGallery-master/" ... and let me know if it works.

**Comment by jamminjames** [comment]
I read the info at https://github.com/Gix075/donkeyGallery, and tried the documentation link, but the site was down at the time. Looking thru the files, I can't find where to set that. Can this plugin be installed on a wordpress site like any other, using the zip file?

**Comment by Gix075** [comment]
Yes the site is temporary down (will be online again on the next days) but you can read some about it on GitHub page. You can set the option on gallery-demo.html
For your last question the answer is no. This is not a WP plugin, this is a package for static sites without cms.

**Comment by jamminjames** [comment]
Okay, I set the subdomain as you suggested, but no change.

**Comment by Gix075** [comment]
Now it's seems that plugin can't write on your disk. If plugin can not generate thubs it will return an ajax error. To be sure that is not a writing permission problem, try to give 775 or 777 permissions to the image folder and its contents.

**Comment by jamminjames** [comment]
That did it, it's working now. Thanks! However, it seems to take a longer time than it should generating the thumbnails, about five seconds to get them all. Is that normal?

**Comment by Gix075** [comment]
Well, it works! Speed depends from the server. Remember to remove all not image files from images folder.

**Comment by foug** [comment]
Hi, similar issue to jamminjames, I've edited my folder permissions but the gallery still sits at the loading icon. I've testing the demo page.

My console in dev tools returns this error, "jquery.donkeyGallery.js:147 Uncaught TypeError: Cannot read property 'jsonFancy' of null"

**Comment by rodrigolinkweb** [comment]
The instructions are not so clearly for me too... I can't understand the difference between these lines:
galleryPath : "dnk-gallery/images/oktoberibs/",
subdomain: "",

I have an index.php, which is the same as gallerydemo.php on the root, then I have a subfolder called dnk-gallery with 4 subfolders inside of if called assets, css, php and js.
When I try, it keeps loading forever. What am I doing wrong, please???!

**Comment by Gix075** [comment]
Sorry for my bad English. I hope this can help you.

**galleryPath**  parameter is the gallery dir path (where you have the gallery pics) relative to the page where gallery will be showed. (In your case relative to index.php).

If the page (or the site) where you want to show the gallery is not at the server root level you need to put the site subdir path inside the **subdomain** parameter.

Sometimes, on some server configuration, you need to put inside **galleryPath**  the absolute path of your gallery dir (something as _/home/user/username.com/public_html_)

**Comment by rodrigolinkweb** [comment]
My structure:
public_html/gallerydemo.php
public_html/dnk-gallery/
public_html/dnk-gallery/assets
public_html/dnk-gallery/css
public_html/dnk-gallery/images (here I have subfolders for the albums)
public_html/dnk-gallery/js
public_html/dnk-gallery/php
public_html/dnk-gallery/php/

my code:
webservice : "dnk-gallery/php/gallery.webservice.php",
galleryPath : "dnk-gallery/images/",
subdomain: "",

Sorry but I didn't understand what you mean... can you help me more?

**Comment by Gix075** [comment]
Reading your structure and your relative cose I can say you that if you have some subfolders inside the main "images" directory, you need to launch plugin one time for each folder. You must think that for each gallery you want to load in your page you need an univocal folder for the gallery images and a new plugin init. 
So, if you charge all galleries inside the main images directory, you need to set the plugin with the option:
```
galleryPath : "dnk-gallery/images/subfoldername/"
```
I hope this can help you.

**Comment by IgnacioGans** [comment]
> json

Same problem, any help?
